widgetssdk/src/main/java/com/glia/widgets/core/callvisualizer/domain/VisitorCodeRepository.kt

```diff
@@ -0,0 +1,32 @@
+package com.glia.widgets.core.callvisualizer.domain
+
+import com.glia.androidsdk.GliaException
+import com.glia.androidsdk.omnibrowse.VisitorCode
+import com.glia.widgets.di.GliaCore
+import io.reactivex.Observable
+
+class VisitorCodeRepository(private val gliaCore: GliaCore) {
+
+    fun getVisitorCode(): Observable<VisitorCode?> {
+        if (!gliaCore.isInitialized) {
+            return Observable.error(
+                GliaException(
+                    "Widgets SDK is not initialized",
+                    GliaException.Cause.INVALID_INPUT
+                )
+            )
+        }
+        // TODO improve for testing, suggestions are welcome
+        return Observable.create<VisitorCode> { emitter ->
+            gliaCore.callVisualizer.getVisitorCode { response: VisitorCode?, exception: GliaException? ->
+                if (response != null && response.code != null) {
+                    emitter.onNext(response)
+                } else if (exception != null) {
+                    emitter.onError(exception)
+                } else {
+                    emitter.onError(IllegalArgumentException("No value presented"))
+                }
```

Comment:
If it comes to the TODO comment above, I would extract the `if..else` logic to a function, mock `emitter`, and verify that appropriate functions are called for different conditions.